pymeasure/instruments/hp/hp8657b.py

```diff
@@ -0,0 +1,200 @@
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from enum import IntEnum
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class HP8657B(Instrument):
+    """ Represents the Hewlett Packard 8657B signal generator
+    and provides a high-level interface for interacting
+    with the instrument.
+    """
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(
+            adapter,
+            "Hewlett-Packard HP8657B",
+            includeSCPI=False,
+            send_end=True,
+            **kwargs,
+        )
+
+    class Modulation(IntEnum):
+        """
+        IntEnum for the different modulation sources
+        """
+        EXTERNAL = 1
+        INT_400HZ = 2
+        INT_1000HZ = 3
+        OFF = 4
+        DC_FM = 5
+
+    def check_errors(self):
+        """
+        Method to read the error status register
+        as the 8657B does not support any readout of values, this will return 0 and log a warning
+
+        """
+        log.warning("HP8657B Does not support error status readout")
+
+    def clear(self):
+        """
+        Reset the instrument to power-on default settings
+
+        """
+        self.adapter.connection.clear()
+
+    id = "HP,8657B,N/A,N/A"  #: Manual ID entry
+
+    am_depth = Instrument.setting(
+        "AM %2.1f PC",
+        """
+        Set the modulation depth for AM,
+        usable range 0-99.9%
+        """,
+        validator=strict_range,
+        values=[0, 99.9],
+        )
+
+    am_source = Instrument.setting(
+        "AM S%i",
+        """
+        Set the source for the AM function
+
+        ==========  =======
+        Value       Meaning
+        ==========  =======
+        OFF         no modulation active
+        Int_400Hz   internal 400 Hz modulation source
+        Int_1000Hz  internal 1000 Hz modulation source
+        External    External source, AC coupling
+        ==========  =======
+
+        *Note:*
+            * AM & FM can be active at the same time
+            * only one internal source can be active at the time
+            * use "OFF" to deactivate AM
+
+        """,
+        validator=strict_discrete_set,
+        values=Modulation,
+        )
```

Comment:
Please give the user a hint, that he might use the `Modulation` enum (similarly below). Otherwise the user will never know.

Reply:
I will add a codeblock exmaplew how to use modulations